src/integrations/watchers/indexable-ancestor-watcher.php

```diff
@@ -124,12 +124,15 @@ public function reset_children( $indexable, $indexable_before ) {
 		$child_indexable_ids = $this->indexable_hierarchy_repository->find_children( $indexable );
 		$child_indexables    = $this->indexable_repository->find_by_ids( $child_indexable_ids );
 
-		\array_walk( $child_indexables, [ $this, 'update_hierarchy_and_permalink' ] );
-
+		if ( ! empty( $child_indexables ) ) {
+			\array_walk( $child_indexables, [ $this, 'update_hierarchy_and_permalink' ] );
+		}
 		if ( $indexable->object_type === 'term' ) {
 			$child_indexables_for_term = $this->get_children_for_term( $indexable->object_id, $child_indexables );
 
-			\array_walk( $child_indexables_for_term, [ $this, 'update_hierarchy_and_permalink' ] );
+			if ( ! empty( $child_indexables_for_term ) ) {
```

Comment:
You are right! Added the check and will look into reproducing the issue

Reply:
I'd remove the `if ( ! empty( $child_indexables_for_term ) ) {` and the `if ( ! empty( $child_indexables ) ) {` added checks since array_walk works perfectly fine with an empty array as well

Reply:
I have reverted both changes :)